seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/source/HttpSourceReader.java

```diff
@@ -153,6 +153,11 @@ private void updateRequestParam(PageInfo pageInfo) {
                 .put(pageInfo.getPageField(), pageInfo.getPageIndex().toString());
     }
 
+    @Override
+    public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
+        internalPollNext(output);
```

Comment:
```suggestion
        synchronized (lock) {
            internalPollNext(output);
        }
```

Reply:
we should lock it to make sure checkpoint not trigger before split read finished.